R/position-boxdodge.r

```diff
@@ -0,0 +1,116 @@
+#' Position dodge for box plots
```

Comment:
Or maybe `position_flexdodge()`, since it's a more flexible dodge that can handle variable width boxes and arbitrary rectangles?